### ISSUE TITLE
ogr2ogr GML->GPKG conversion via .gfs - can't map attributes AND values when xsi:nil='true' present

### ISSUE THREAD

**Issue by JohnNewland** [OSGeo/gdal#1754]
When using a .gfs to map GML->GPKG, using INSPIRE GML source data, it is possible for the same field to have an attribute value in one feature and a value in another feature (without the attribute). It seems to happen especially with xsi:nil and nilReason attributes

e.g. 
`feature 1: <test:aField>a value</test:aField>`
`feature 2: <test:aField xsi:nil="true" nilReason="inapplicable"/>`
...are possible for two different features

With the provided test data and .gfs mapping, the results are as follows:

                   aField          aFieldNilReason
feature 1    a value                null
feature 2    null                     null

...whereas, we would like to see:

                   aField          aFieldNilReason
feature 1    a value           null
feature 2    null                inapplicable

...or ideally, but perhaps not possible (by mapping both attributes and values into the same field):

                    aField          
feature 1    a value            
feature 2    inapplicable

It seems as though we cannot map an ElementPath of aField and an ElementPath of aField@attribute in the same .gfs file while in the presence of xsi:nil='true', because when you remove the aField mapping, it maps aField@attribute just fine
Interestingly, when you remove xsi:nil="true" from the second feature, leaving nilReason - it works as desired:

                  aField          aFieldNilReason
feature 1    a value           null
feature 2    null                inapplicable

I can't find any option on the GML driver that could help us here and we would prefer to not use the GMLAS driver as we had many other issues with that.

I have linked #1154  as this (and the enclosed linked issue from another github repo) explicitly mention that you cannot retrieve attributes when an xsi:nil attribute is involved, but I seem to be able to in certain circumstances using the .gfs mapping method.

Environment:
OS: Windows NT 6.2 (Server 2012)
GDAL version: GDAL 2.4.0, released 2018/12/14

Example data:
[example.zip](https://github.com/OSGeo/gdal/files/3477162/example.zip)

Example command line:
ogr2ogr -progress --config GML_GFS_TEMPLATE example.gfs -dsco VERSION=1.2 --config GML_EXPOSE_GML_ID YES -f GPKG example.gpkg example.gml -a_srs EPSG:27700 -lco SPATIAL_INDEX=NO

Thank you.

**Comment by jratike80** [comment]
Out of interest, could you summarize the many other issues you had with the GMLAS driver? After all that's the primary driver for handling complex GML.

**Comment by JohnNewland** [comment]
> Out of interest, could you summarize the many other issues you had with the GMLAS driver? After all that's the primary driver for handling complex GML.

Unfortunately, the team member who dealt with this is off for a couple of weeks; I'd need to check with her why she decided that we couldn't use this driver. 
I gave it a go briefly and it couldn't resolve many of the online .xsd locations, despite them being clearly accessible from the machine. I would bring our schema locally to get around the first issue, but then it calls out to many other online schema such as those from w3.org and fails to resolve those.

**Comment by jratike80** [comment]
I made a quick test with `ogrinfo gmlas:example.gml --debug on
`  and I got the following error:
```
ERROR 1: example.gml:2:493 schemaLocation does not contain namespace-location pairs
ERROR 1: No schema locations found when analyzing data file: XSD open option must be provided
```

I do not really understand XML but the error seems to be caused by the end of the second line 
`xsi:schemaLocation="http://test.co.uk/Namespacesns.xsd"`. I fear I can't help you much by making tests with GMLAS because it is so strict with validating XML and schemas. But I have used GMLAS driver with INSPIRE data and the driver should be able to resolve online schemas if they are defined right in XML, if your proxy server allows you to reach them, and if schema files are online.

**Comment by JohnNewland** [comment]
Ah - don't worry about the test data - it is just that. I wouldn't put the real data up here. Thanks very much for your help.
